core/src/main/java/org/elasticsearch/script/ScriptType.java

```diff
@@ -24,66 +24,117 @@
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
-import java.util.Locale;
+import java.util.Objects;
 
 /**
- * The type of a script, more specifically where it gets loaded from:
- * - provided dynamically at request time
- * - loaded from an index
- * - loaded from file
+ * ScriptType represents the way a script is stored and retrieved from the {@link ScriptService}.
+ * It's also used to by {@link ScriptSettings} and {@link ScriptModes} to determine whether or not
+ * a {@link Script} is allowed to be executed based on both default and user-defined settings.
  */
 public enum ScriptType {
 
-    INLINE(0, "inline", "inline", false),
-    STORED(1, "id", "stored", false),
-    FILE(2, "file", "file", true);
+    /**
+     * INLINE scripts are specified in numerous queries and compiled on-the-fly.
+     * They will be cached based on the lang and code of the script.
+     * They are turned off by default for security purposes.
```

Comment:
They are turned off for groovy by default because it is insecure but they are on by default for painless and mustache.